cmd/juju/cloud/addcredential.go

```diff
@@ -26,41 +26,74 @@ var usageAddCredentialSummary = `
-credentials, of which there may be multiple per cloud.
-The ` + "`--replace`" + ` option is required if credential information for the named
-cloud already exists locally. All such information will be overwritten.
-This command does not set default regions nor default credentials. Note
-that if only one credential name exists, it will become the effective
-default credential.
-For credentials which are already in use by tools other than Juju, ` + "`juju \nautoload-credentials`" + ` may be used.
-When Juju needs credentials for a cloud, i) if there are multiple
-available; ii) there's no set default; iii) and one is not specified ('--
-credential'), an error will be emitted.
+juju add-credential operates in two modes.
+
+When called with only the <cloud name> argument, ` + "`juju add-credential`" + ` will 
+take you through an interactive prompt to add a credential specific to 
+the cloud provider.
+
+Providing the ` + "`-f <credentials.yaml>` " + `option switches to the 
+non-interactive mode. <credentials.yaml> must be a path to a correctly 
+formatted YAML-formatted file. Details of the format are provided at 
+"About credentials.yaml" below.
+
+The ` + "`--replace`" + ` option is required if credential information 
+for the named cloud already exists locally. All such information will be 
+overwritten.
+
+About credentials.yaml:
+Here is a sample credentials.yaml showing four credentials being stored 
+against three clouds:
+
+  credentials:
+    aws:
+      <credential-name>:
+        auth-type: access-key
+        access-key: <key>
+        secret-key: <key>
+    azure:
+      <credential-name>:
+        auth-type: service-principal-secret
+        application-id: <uuid>
+        application-password: <password>
+        subscription-id: <uuid>
+    lxd:
+      <credential-name>:
+        auth-type: interactive
+        trust-password: <password>
+      <credential-name>:
+        auth-type: interactive
+        trust-password: <password>
+
+More generally, here is a loosely defined grammar for credentials.yaml:
+
+  credentials:
+    <cloud-name>:
+      <credential-name>:
+        auth-type: <auth-type>
+        <auth-type-key>: <auth-type-value>
+        [<auth-type-key>: <auth-type-value>]
+
+Every <auth-type> requires its own <auth-type-key> and <auth-type-value>
+pairs. 
+
+The <credential-name> parameter of each credential is arbitrary, but must
+be unique within each <cloud-name>. This allows allow each cloud to store 
+multiple credentials.
 
 Examples:
     juju add-credential google
     juju add-credential aws -f ~/credentials.yaml
+    juju add-credential aws --replace -f ~/credentials.yaml
+
+Notes:
+If you setting up Juju for the first time, consider running
+` + "`juju autoload-credentials`" + `. This may help you skip adding credentials
+manually.
+
+This command does not set default regions nor default credentials for the
+cloud. The commands ` + "`juju set-default-region`" + ` and `+ "`juju set-default-region`" + `
```

Comment:
These should be "See also" as well, plus looks like there's a type (dup command)

Reply:
Good point. Have included it.